spec/models/environment_spec.rb

```diff
@@ -16,17 +27,33 @@
   describe '#default_ref' do
     context 'when the environment has a default ref provided' do
       it 'returns that value' do
-        environment = Environment.new(default_ref: 'develop')
+        repo = Repository.with_name('acme-inc/api')
+        environment = Environment.new(default_ref: 'develop', repository: repo)
         expect(environment.default_ref).to eq 'develop'
       end
     end
 
     context 'when the environment does not have a default ref' do
       it 'returns the global default' do
-        environment = Environment.new(default_ref: '')
+        repo = Repository.with_name('acme-inc/api')
+        environment = Environment.new(default_ref: '', repository: repo)
         expect(environment.default_ref).to eq 'master'
 
-        environment = Environment.new
+        environment = Environment.new(repository: repo)
+        expect(environment.default_ref).to eq 'master'
+      end
+    end
+    context 'when the repository has a config set' do
+      let(:octokit_client) { instance_double(Octokit::Client) }
+      let(:user) { stub_model(User, octokit_client: octokit_client) }
+      it 'returns the default_ref from the yaml file' do
+        repo = Repository.with_name('acme-inc/api')
+        repo.configure! <<-YAML
+          environments:
+            production:
+              aliases: [prod]
+          YAML
+        environment = Environment.new(name: 'production', repository: repo)
```

Comment:
The big change here is that we are updating this test so that an environment always has a repo attached. If we don't, our method fails. I can't think of a scenario where we have a lone environment not connected to a repo, but correct me if that's not the case.